gconf/genesis_test.go

```diff
@@ -13,7 +14,10 @@ func TestGenesisInitializer(t *testing.T) {
 		{
 			"gconf": {
 				"a-string": "hello",
-				"an-int": 321
+				"an-int": 321,
+				"string-coin": "4 IOV",
+				"struct-coin": {"whole": 7, "ticker": "IOV"},
+				"an-address": "d2a1f84143a9754057e42db6d6c9f986fe0ff673"
```

Comment:
This is a test that I should have updated as the first thing because it shows that this approach makes no sense.

We are using JSON to represent `gconf` values in a genesis file. The problem is that JSON types do not match our types. This is not an issue for numbers - I can cast flaot64 to int64 becasue a JSON number represents only one type - int64. This is not possible with strings. 
A string in our JSON configuration can represent multiple things
- a hex encoded `[]byte`
- a hex encoded `weave.Address`
- a `coin.Coin` instance
- a `string` value

I could do some assumptions and try to additionally parse a string, but that is not good enough. For example a coin written as `"4 IOVVV"` is not valid and therefore it will be stored an gconf string.

Unless a type information or a schema is provided for gconf's genesis representation, this task is not doable.